subsys/bluetooth/host/CMakeLists.txt

```diff
@@ -111,6 +111,6 @@ endif()
 # Bluetooth Mesh has test dependencies in the host.
 # In order to compile Bsim tests with these test features
 # and PSA enabled, the libraries must be linked.
-if(CONFIG_BT_MESH AND CONFIG_BT_TESTING)
+if((CONFIG_BT_MESH AND CONFIG_BT_TESTING) OR CONFIG_BT_USE_PSA_API)
```

Comment:
The same question here. Why is it required if TF-M is used?
This was added for mesh<->host dependencies for BabbleSim tests. BabbleSim tests are built for POSIX based platforms. The only mbedTLS is supported for them. 
Since your PR adds PSA support in Host for both pure mbedTLS and TF-M I guess condition here should consider the PSA implementation provider.